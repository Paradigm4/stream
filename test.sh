#!/bin/bash

MYDIR=`dirname $0`
pushd $MYDIR > /dev/null
MYDIR=`pwd`

iquery -aq "remove(foo)" > /dev/null 2>&1
rm -rf $MYDIR/test.out

iquery -anq "store(build(<val:double>[i=1:800000,100000,0], i), foo)" > /dev/null 2>&1
iquery -aq "op_count(stream(foo, '$MYDIR/examples/stream_test_client'))" >> $MYDIR/test.out 2>&1
iquery -aq "stream(foo, '$MYDIR/examples/stream_test_client')" | head -n 5 >> $MYDIR/test.out 2>&1
iquery -otsv -aq "stream(_sg(foo, 2,0), '$MYDIR/examples/stream_test_client SUMMARIZE')" | head -n 1 >> $MYDIR/test.out 2>&1

#Test READ_DELAY: start a client with a delay
iquery -aq "op_count(stream(foo, '$MYDIR/examples/stream_test_client READ_DELAY'))" > /dev/null 2>&1 &
sleep 1
#Find the query and cancel it
iquery -aq "op_count(filter(list('queries'), inst=0))" >> $MYDIR/test.out 2>&1
QID=`iquery -otsv -aq "filter(list('queries'), inst=0 )" | grep stream | awk '{print $1}'`
iquery -aq "cancel('$QID')" >> $MYDIR/test.out 2>&1
sleep 5
#After 5 seconds, better be cleared up
iquery -aq "op_count(filter(list('queries'), inst=0))" >> $MYDIR/test.out 2>&1

#Test WRITE_DELAY: start a client with a delay
iquery -aq "op_count(stream(foo, '$MYDIR/examples/stream_test_client WRITE_DELAY'))" > /dev/null 2>&1 &
sleep 1
#Find the query and cancel it
iquery -aq "op_count(filter(list('queries'), inst=0))" >> $MYDIR/test.out 2>&1
QID=`iquery -otsv -aq "filter(list('queries'), inst=0 )" | grep stream | awk '{print $1}'`
iquery -aq "cancel('$QID')" >> $MYDIR/test.out 2>&1
sleep 5
#After 5 seconds, better be cleared up
iquery -aq "op_count(filter(list('queries'), inst=0))" >> $MYDIR/test.out 2>&1

iquery -ocsv -aq "stream(build(<val:string> [i=1:10,5,0], i), 'Rscript $MYDIR/examples/tsv_R_client.R')" >> $MYDIR/test.out 2>&1
iquery -aq "op_count(stream(foo, 'Rscript $MYDIR/examples/tsv_R_client.R'))" >> $MYDIR/test.out 2>&1

iquery -aq "stream(build(<val:double> [i=1:5,5,0], i), 'Rscript $MYDIR/examples/R_client.R', 'format=df', 'types=double,int32')" >> $MYDIR/test.out 2>&1

iquery -aq "
 stream(
  _sg(
   stream(
    build(<val:double> [i=1:50,10,0], i), 
    'Rscript $MYDIR/examples/R_sum.R', 
    'format=df', 
    'types=double', 
    'names=sum'
   ), 
   2, 0
  ), 
  'Rscript $MYDIR/examples/R_sum.R', 
  'format=df', 
  'types=double', 
  'names=sum'
 )" >> $MYDIR/test.out 2>&1

iquery -aq "stream(build(<val:string> [i=1:10,10,0], iif(i=1,'', string(i))), 'Rscript $MYDIR/examples/R_strings.R', 'format=df', 'types=string', 'names=s')" >> $MYDIR/test.out 2>&1

iquery -aq "stream(
 apply(
  build(<a:int32>[i=1:3,3,0], i), 
  dub, double(iif(a=1, null, iif(a=2, 0,  1))),  
  i32, int32( iif(a=1, null, iif(a=2, 0,  1))), 
  str,        iif(a=1, null, iif(a=2, '', 'abc'))
 ), 
 'Rscript $MYDIR/examples/R_identity.R', 'format=df', 'types=int32,double,int32,string', 'names=a,b,c,d'
)" >> $MYDIR/test.out 2>&1

diff test.expected test.out

