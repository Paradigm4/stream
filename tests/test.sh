#!/bin/bash

# These tests assumes a 2 instance configuration.  A different number
# of instances yield different numbers of computations from the
# stream() operator, which executes on all instances.

MY_DIR=`dirname $0`
pushd $MY_DIR > /dev/null
MY_DIR=`pwd`
EX_DIR=`pwd`/../examples

iquery -aq "remove(foo)" > /dev/null 2>&1
rm -rf $MY_DIR/test.out

iquery -anq "store(build(<val:double>[i=1:800000:0:100000], i), foo)" > /dev/null 2>&1
iquery -aq "op_count(stream(foo, '$EX_DIR/stream_test_client'))" >> $MY_DIR/test.out 2>&1
iquery -aq "stream(foo, '$EX_DIR/stream_test_client')" | head -n 5 >> $MY_DIR/test.out 2>&1
iquery -otsv -aq "stream(_sg(foo, 2,0), '$EX_DIR/stream_test_client SUMMARIZE')" | head -n 1 >> $MY_DIR/test.out 2>&1

#Test READ_DELAY: start a client with a delay
iquery -aq "op_count(stream(foo, '$EX_DIR/stream_test_client READ_DELAY'))" > /dev/null 2>&1 &
sleep 1
#Find the query and cancel it
iquery -aq "op_count(filter(list('queries'), inst=0))" >> $MY_DIR/test.out 2>&1
QID=`iquery -otsv -aq "filter(list('queries'), inst=0 )" | grep stream | awk '{print $1}'`
iquery -aq "cancel('$QID')" >> $MY_DIR/test.out 2>&1
sleep 5
#After 5 seconds, better be cleared up
iquery -aq "op_count(filter(list('queries'), inst=0))" >> $MY_DIR/test.out 2>&1

#Test WRITE_DELAY: start a client with a delay
iquery -aq "op_count(stream(foo, '$EX_DIR/stream_test_client WRITE_DELAY'))" > /dev/null 2>&1 &
sleep 1
#Find the query and cancel it
iquery -aq "op_count(filter(list('queries'), inst=0))" >> $MY_DIR/test.out 2>&1
QID=`iquery -otsv -aq "filter(list('queries'), inst=0 )" | grep stream | awk '{print $1}'`
iquery -aq "cancel('$QID')" >> $MY_DIR/test.out 2>&1
sleep 5
#After 5 seconds, better be cleared up
iquery -aq "op_count(filter(list('queries'), inst=0))" >> $MY_DIR/test.out 2>&1

iquery -ocsv -aq "stream(build(<val:string> [i=1:10:0:5], i), 'Rscript $EX_DIR/tsv_R_client.R')" >> $MY_DIR/test.out 2>&1
iquery -aq "op_count(stream(foo, 'Rscript $EX_DIR/tsv_R_client.R'))" >> $MY_DIR/test.out 2>&1

iquery -aq "remove(foo)" > /dev/null 2>&1

iquery -ocsv -aq "stream(build(<val:double> [i=1:5:0:5], i), 'Rscript $EX_DIR/R_client.R', format:'df', types:('double','int32'))" >> $MY_DIR/test.out 2>&1

iquery -aq "
 stream(
  _sg(
   stream(
    build(<val:double> [i=1:50:0:10], i),
    'Rscript $EX_DIR/R_sum.R',
    format:'df',
    types:'double',
    names:'sum'
   ),
   2, 0
  ),
  'Rscript $EX_DIR/R_sum.R',
  format:'df',
  types:'double',
  names:'sum'
 )" >> $MY_DIR/test.out 2>&1

iquery -ocsv -aq "stream(build(<val:string> [i=1:10,10,0], iif(i=1,'', string(i))), 'Rscript $EX_DIR/R_strings.R', format:'df', types:'string', names:'s')" >> $MY_DIR/test.out 2>&1

iquery -ocsv -aq "stream(
 apply(
  build(<a:int32>[i=1:3:0:3], i),
  dub, double(iif(a=1, null, iif(a=2, 0,  1))),
  i32, int32( iif(a=1, null, iif(a=2, 0,  1))),
  str,        iif(a=1, null, iif(a=2, '', 'abc'))
 ),
 'Rscript $EX_DIR/R_identity.R', format:'df', types:('int32','double','int32','string'), names:('a','b','c','d'))" >> $MY_DIR/test.out 2>&1

#Conversion from client->scidb and then scidb->iquery adds extra backslashes; bear with us!
iquery -otsv -aq "stream(apply(build(<a:string> [i=0:0:0:1], '\n \r \t \ '), b, string(null)), '$EX_DIR/stream_test_client')" >> $MY_DIR/test.out 2>&1

iquery -ocsv -aq "stream(build(<a:double>[i=0:9:0:10],i), 'python $EX_DIR/python_example.py')" >> $MY_DIR/test.out 2>&1

diff $MY_DIR/test.expected $MY_DIR/test.out
