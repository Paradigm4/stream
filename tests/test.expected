{i} count
{0} 8
{instance_id,chunk_no} response
{0,0} 'Hello	100001
Hello	100002
Hello	100003
Hello	100004
Thanks! That was a total of 800000 lines.
{i} count
{0} 2
Query was executed successfully
{i} count
{0} 1
{i} count
{0} 2
Query was executed successfully
{i} count
{0} 1
'6	12
7	14
8	16
9	18
10	20'
'1	2
2	4
3	6
4	8
5	10'
{i} count
{0} 8
2,8
4,16
6,24
8,32
10,40
{instance_id,chunk_no,value_no} sum
{0,0,0} 1275
'Hello, !'
'Hello, 2!'
'Hello, 3!'
'Hello, 4!'
'Hello, 5!'
'Hello, 6!'
'Hello, 7!'
'Hello, 8!'
'Hello, 9!'
'Hello, 10!'
''
'Thx!'
1,null,null,null
2,0,0,''
3,1,1,'abc'
Hello\t\\n \\r \\t \\\\ \t\\N\nOK\tthanks!
'KTHXBYE'
'I got	0
I got	1
I got	2
I got	3
I got	4
I got	5
I got	6
I got	7
I got	8
I got	9
THX!'
'KTHXBYE'
